src/coreclr/jit/fgopt.cpp

```diff
@@ -3712,89 +3596,45 @@ bool Compiler::fgReorderBlocks(bool useProfile)
                     // We will consider all blocks that have less weight than profHotWeight to be
                     // uncommonly run blocks as compared with the hot path of bPrev taken-jump to bDest
                     //
-                    if (fgHaveValidEdgeWeights)
+                    // We will check that the weight of the bPrev to bDest edge
+                    //  is more than twice the weight of the bPrev to block edge.
+                    //
+                    //                  bPrev -->   [BB04, weight 31]
+                    //                                     |         \.
+                    //          edgeToBlock -------------> O          \.
+                    //          [min=8,max=10]             V           \.
+                    //                  block -->   [BB05, weight 10]   \.
+                    //                                                   \.
+                    //          edgeToDest ----------------------------> O
+                    //          [min=21,max=23]                          |
+                    //                                                   V
+                    //                  bDest --------------->   [BB08, weight 21]
+                    //
+                    assert(bPrev->FalseTargetIs(block));
+                    FlowEdge* edgeToDest  = bPrev->GetTrueEdge();
+                    FlowEdge* edgeToBlock = bPrev->GetFalseEdge();
+                    noway_assert(edgeToDest != nullptr);
+                    noway_assert(edgeToBlock != nullptr);
+                    //
+                    // Calculate the taken ratio
+                    //   A takenRatio of 0.10 means taken 10% of the time, not taken 90% of the time
+                    //   A takenRatio of 0.50 means taken 50% of the time, not taken 50% of the time
+                    //   A takenRatio of 0.90 means taken 90% of the time, not taken 10% of the time
+                    //
+                    double takenCount    = edgeToDest->getLikelyWeight();
```

Comment:
We should revise this to just use likelihoods, instead of weights? But we can do this as a  follow-up.